### PR TITLE
Better handling of header in snapshots

### PR DESCRIPTION
**Pull request by hadley**

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -331,7 +331,7 @@ header <- function(label,
                    suffix = "-",
                    n = min(getOption("width"), 78))
 {
-  n <- n - nchar(label) - nchar(prefix) - 2L
+  n <- max(n - nchar(label) - nchar(prefix) - 2L, 3)
```
Comment (kevinushey) [inline R/utils.R:334]: This ensures there will always be at least three `---` following the header text?

**Review by kevinushey — APPROVED** [review]
LGTM!

File: R/utils.R (hunk 1 above)
Comment (hadley) [reply R/utils.R:334]: Yeah, I don't think it makes much difference for real life, but guarantees that there's something unique for the transformer to latch on to.

File: tests/testthat/helper-snapshot.R [hunk 2]
```diff
@@ -45,6 +45,9 @@ strip_dirs <- function(x) {
   x <- gsub(renv_path_aliased(getwd()), "<wd>", x, fixed = TRUE)
   x <- gsub(renv_path_aliased(tempdir()), "<tempdir>", x, fixed = TRUE)
 
+  # Standardise the dashes produced by header()
+  x <- gsub("-{3,}", "---", x, fixed = TRUE)
```
Comment (hadley) [inline tests/testthat/helper-snapshot.R:49]: I meant this to be `perl = TRUE` 😬

**Pull request merged**